Tutorial: Transfer Getting Started Tutorial from community forum

DocFraggle opened the pull request:
https://community.openhab.org/t/setting-up-openhab2-from-scratch/20080

Hi Thomas,

here is the PR for my tutorial. Please have a look if it's ok this way!

Regards, Christian

ThomDietrich commented:
That was quick! I will start reviewing right away, not sure if I'll finish today though. Enjoy your evening 😉

DocFraggle commented:
Hmm... no idea why this merge appeared here, I just synced my fork with the upstream repo...

ThomDietrich commented:
Because you are working in the main branch (here: gh-pages). Never start a Pull Request from the main branch of your fork. It's okay now, you'll know that next time ;)
Please squash all commits in this PR and sign your work before we are going to merge it.

DocFraggle commented:
Yeah, got confused there because there was no master branch :) should be ok now. Should I address the comments of Jörg before you'll merge this PR? Or should I do that afterwards?

ThomDietrich commented:
Yes please do. I will add my own share of comments this evening.

kaikreuzer commented on tutorials/beginner/1sttimesetup.md:
```diff
@@ -0,0 +1,22 @@
+
+{% include base.html %}
+
+# First-time setup of openHAB2
```
I would suggest to replace all occurrences of "openHAB2" and "openHAB 2" by simply "openHAB".

kaikreuzer commented on tutorials/beginner/configuration.md:
```diff
@@ -0,0 +1,122 @@
+
+![](images/picture_10.jpg)
+
+_Note: Thing names have to be unique! You should follow a certain naming convention concerning your things. In this example, we used the purpose of the thing, "Presence", followed by the device type, "_Mobile", and finally the owner's name, "_John", to contruct the final thing name. Another suggestion for different things, for example a wallplug which is plugged in in the living room in first flo ...
```
> Thing names have to be unique!

No, actually not! Things have a technical UID, which is unique. The name that the user can give it here should not be technical, but a label which is nice to show on UIs and which means something to the user. Therefore it should NOT use any underscore notation, but can have spaces and other special characters (e.g. Umlauts, etc.).

ThomDietrich replied:
We've been talking about naming conventions but in the context of item definitions. There all the mentioned notation notes are still valid, so you might want to copy them there.

@kaikreuzer I've been talking to @DocFraggle via PM about this tutorial and we should talk about naming conventions and Style guides. Not here not now. openHAB should provide that as guidance.

For this tutorial we agreed to stay with the Demo setup config file style, hence e.g. Wallplug_FF_LR_TV

kaikreuzer replied:
You are indeed mixing item names with the thing labels, which is not good.

This "Note" paragraph is simply wrong, so it needs to be corrected. If the tutorial uses for now an underscore syntax is not nice, but ok, probably it cannot changed too easily. But the risk is that people actually think this is how Things have to be named and do it that way forever...

ThomDietrich replied:
I am not talking about things. Thing labels as well as item labels should hold a meaningful name without underscore and such.

I agree with your fist comment but wanted to justify the idea behind the note and pointed out that this whole paragraph can be moved (instead of being deleted) over to the part where items are created manually in files.

DocFraggle replied:
Hi, before I change the thing names again, I just want to check if these thing labels are OK:

"John's Mobile"
"Wallplug TV"

I won't touch the items though, they're still

"Presence_Mobile_John"
"Wallplug_FF_LR_TV"

OK this way?

kaikreuzer replied:
Perfect 👍

kaikreuzer commented on tutorials/beginner/uis.md:
```diff
@@ -6,25 +6,25 @@ layout: tutorial-beginner
 
 # OpenHAB 2 UIs
```
change to lowercase openHAB

kaikreuzer commented on tutorials/beginner/uis.md:
```diff
@@ -6,25 +6,25 @@ layout: tutorial-beginner
 # OpenHAB 2 UIs
 
-OpenHAB 2 offers different UIs in its standard configuration: the _Paper UI_, the _Basic UI_ and the _Classic UI_.
+OpenHAB 2 offers different UIs in its standard configuration: the _Paper UI_, the _Basic UI_ and the _Classic UI_ (this has to be installed manually from within Paper UI or via config file).
```
change to lowercase openHAB

ThomDietrich commented:
@DocFraggle I'm really sorry I didn't find the time to review your PR yet. I will try to get to that today or tomorrow evening.

DocFraggle commented:
Ok, I'll change that for the things and nice this part over to the items!

Christian

-------------------------
VON: Thomas Dietrich <notifications@github.com>
GESENDET: 18.01.2017 15:51
AN: openhab-docs@noreply.github.com
CC: DocFraggle; Mention
BETREFF: Re: [openhab/openhab-docs] Tutorial: Transfer Getting Started
Tutorial from community forum (#207)

> @THOMDIETRICH commented on this pull request.
>
> -------------------------
>
> In tutorials/beginner/configuration.md[1]:
>> + +* add the device as a thing by clicking the check icon +* ignore the
>> device by clicking the crossed-out-eye icon +* delete the device by
>> clicking the trashcan icon + +The difference between ignoring and
>> deleting discovered things is as follows: + +* if you ignore a thing,
>> it disappears from the visible inbox. If you decide to add it at a
>> later time, you can do so by scrolling to the bottom of the inbox and
>> click the "SHOW IGNORED" button. +* if you delete the thing, it won't
>> be visible in the inbox any more. But if you start a new discovery
>> process later, the thing will reappear. + +In this example, we click on
>> the check icon of the thing with the IP address 192.168.1.103. Let's
>> assume this is John's mobile phone which we want to use for presence
>> detection. We change the (automatically added during the discovery)
>> name of the thing on the next page to "Presence_Mobile_John" and click
>> on the "ADD AS THING" button. +_Please note: if your network devices
>> use a DHCP server to obtain ther IP addresses automatically, please
>> make sure to define a DHCP reservation for a device which you want to
>> use in openHAB 2. Otherwise it's IP address may change in the future
>> and your thing definiton won't work any more.  Information about how to
>> define a DHCP reservation can be normally be found in the manual of
>> your network router._ + +![](images/picture_10.jpg) + +_Note: Thing
>> names have to be unique! You should follow a certain naming convention
>> concerning your things. In this example, we used the purpose of the
>> thing, "Presence", followed by the device type, "_Mobile", and finally
>> the owner's name, "_John", to contruct the final thing name. Another
>> suggestion for different things, for example a wallplug which is
>> plugged in in the living room in first floor providing power to a TV,
>> is to use the type, the floor, the room and the usage of the thing. In
>> this case it would be "Wallplug_FF_LR_TV"._
> I am not talking about things. Thing labels as well as item labels
> should hold a meaningful name without underscore and such.
>
> I agree with your fist comment but wanted to justify the idea behind the
> note and pointed out that this whole paragraph can be moved (instead of
> being deleted) over to the part where items are created manually in
> files.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub[1], or mute the
thread[2].
>
>
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/openhab/openhab-docs","title":"openhab/openhab-docs","subtitle":"GitHub
repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open
in
GitHub","url":"https://github.com/openhab/openhab-docs"}},"updates":{"snippets":[{"icon":"PERSON","message":"@ThomDietrich
commented on #207"}],"action":{"name":"View Pull
> Request","url":"https://github.com/openhab/openhab-docs/pull/207"}}}



Links:
------
[1] https://github.com/openhab/openhab-docs/pull/207
[2]  
https://github.com/notifications/unsubscribe-auth/AXs3GEW0_SZZz8DEG0CVYqxE9IskySD6ks5rTibXgaJpZM4Lj9j4

DocFraggle commented:
OK, I pushed the changes

- renamed [oO]penHAB[ 2] to openHAB
- switched over the naming convention to the sitemap file
- updated the screenshots with the thing names we discussed

Regards, Christian

DocFraggle commented:
@ThomDietrich Hi Thomas, is this OK so far? Anything else I should change?

Regards, Christian

ThomDietrich commented:
Hey Christian,
I did *not* find the time to have a look yet. I was swamped with a lot of other things this week. Sorry.

Would you be okay with the following suggestion: You already received a bit of feedback. We could just as well go ahead and merge your PR right now. You could than immediately open a new PR and we can discuss and step-by-step improve the Tutorial there. This way we will have a first (already pretty good!) version of the tutorial online, which in my opinion is the important thing right now. What do you think?

DocFraggle commented:
Sure, go ahead! :)

ThomDietrich commented:
I've taken a quick look and there is nothing critical from stopping us. Here we go.

Thanks for your work! This will be one of the most important additions to docs for the upcoming 2.0 release!
